frontend/src/modules/activity/components/activity-timeline.vue

```diff
@@ -319,6 +319,7 @@ watch(query, (newValue, oldValue) => {
 
 watch(platform, async (newValue, oldValue) => {
   if (newValue !== oldValue) {
+    offset.value = 0
```

Comment:
This line fixes the described issue. However, I think it might be better to add it on line 265 which is where we have the logic to reset the request
So it would be 
```
  if (!isEqual(filter, filterToApply)) {
    activities.length = 0;
    offset.value = 0
    noMore.value = false;
  }
```
When platform changes the fetch method enters this condition and resets the values. Could you please make this change?